dbschema/product-progress.esdl

```diff
@@ -0,0 +1,19 @@
+module default {
+  type ProductProgress extending Mixin::Timestamped {
+    product: Product;
+    report: PeriodicReport;
+    variant: VariantProgress::Variant;
+    multi steps: StepProgress;
+  }
+  type StepProgress extending Mixin::Timestamped {
+    step: Product::Step;
```

Comment:
This is required and should be exclusive to each `ProductProgress`. The best way is to flip the relation and do the same thing as above.
```suggestion
			 required productProgress: ProductProgress;
    required step: Product::Step;
    
    constraint exclusive on ((.productProgress, .step));
```

Then the `ProductProgress` references the steps with a backlink.
```diff
- multi steps: StepProgress;
+ steps := .productProgress[is StepProgress];
```